chainerx_cc/chainerx/cuda/cuda_device/linalg.cu

```diff
@@ -409,5 +479,58 @@ public:
 
 CHAINERX_CUDA_REGISTER_KERNEL(SvdKernel, CudaSvdKernel);
 
+class CudaSyevdKernel : public SyevdKernel {
+public:
+    void Call(const Array& a, const Array& w, const Array& v, char uplo, bool compute_v) override {
+        Device& device = a.device();
+        Dtype dtype = a.dtype();
+        CudaSetDeviceScope scope{device.index()};
+
+        CHAINERX_ASSERT(a.ndim() == 2);
+
+        device.backend().CallKernel<CopyKernel>(a, v);
+
+        int64_t m = a.shape()[0];
+        int64_t n = a.shape()[1];
+
+        auto syevd_impl = [&](auto pt) {
+            using T = typename decltype(pt)::type;
+            cuda_internal::DeviceInternals& device_internals = cuda_internal::GetDeviceInternals(static_cast<CudaDevice&>(device));
+
+            auto v_ptr = static_cast<T*>(internal::GetRawOffsetData(v));
+            auto w_ptr = static_cast<T*>(internal::GetRawOffsetData(w));
+
+            cusolverEigMode_t jobz = compute_v ? CUSOLVER_EIG_MODE_VECTOR : CUSOLVER_EIG_MODE_NOVECTOR;
+
+            // cuSOLVER assumes that arrays are stored in column-major order
+            // The uplo argument is swapped instead of transposing the input matrix
+            cublasFillMode_t uplo_cublas = (uplo == 'U') ? CUBLAS_FILL_MODE_LOWER : CUBLAS_FILL_MODE_UPPER;
+
+            int buffersize = 0;
+            // When calling Syevd matrix dimensions are swapped instead of transposing the input matrix
+            device_internals.cusolverdn_handle().Call(SyevdBuffersize<T>, jobz, uplo_cublas, n, v_ptr, m, w_ptr, &buffersize);
+
+            Array work = Empty(Shape{buffersize}, dtype, device);
+            auto work_ptr = static_cast<T*>(internal::GetRawOffsetData(work));
+
+            std::shared_ptr<void> devInfo = device.Allocate(sizeof(int));
```

Comment:
We should probably use `device_info` over `deviceInfo` but I noticed similar code is already merged. We can fix them altogether at a later time, unless you want to send a patch and fix the ongoing ones.

Reply:
> unless you want to send a patch and fix the ongoing ones.

I will do this